dvc/utils/fs.py

```diff
@@ -20,15 +20,21 @@ def get_inode(path):
     return inode
 
 
-def get_mtime_and_size(path):
-    stat = os.stat(path)
-    size = stat.st_size
-    mtime = stat.st_mtime
+def get_mtime_signature_and_size(path, ignore_file_handler=None):
+    base_stat = os.stat(path)
+    size = base_stat.st_size
 
     if os.path.isdir(path):
-        for root, dirs, files in dvc_walk(path):
-            for name in dirs + files:
-                entry = os.path.join(root, name)
+        files_mtimes = {}
+        for root, dirs, files in dvc_walk(
+            str(path), ignore_file_handler=ignore_file_handler
+        ):
+            for dir in dirs:
+                entry = os.path.join(root, dir)
+                size += os.path.getsize(entry)
```

Comment:
Why do we care about dir sizes? This looks like a bug, doesn't dir size depend on number of files, including ignored ones?

Reply:
It used to be this way, so we need to keep it to not trigger checksum-recomputation :slightly_frowning_face: Let's add a note though.

Reply:
Actually it only adds constant size. It returns default size for block, in ubuntu case 4096. I guess we could skip that.  

In hypothetical case, where we move some files out of directory and delete it mtime will still change, so change will be detected. I think we don't have to add driectories size to the result.

@efiop Do you see any flaw in my reasoning?

Reply:
Though, it looks like we are changing the way we compute them anyway(e.g. we are not including dir mtimes at all), so we will trigger re-computations anyway. Is that correct, @pared ?

Reply:
ahh sorry, slowpoke

Reply:
On the other side, with new mtime computation way, that is a perfect moment to change way of gettin size, due to new way of mtime calculation, recalculation will be inevitable.

Reply:
slowpoke again